### PR TITLE
DEVX-1967 and DEVX-1279: Migrate Java and C README from Markdown to RST

### PR DESCRIPTION
**Pull request by tikimims**
Converted Java and C README files from Markdown to RST

### CODE REVIEW

File: clients/docs/java.rst [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+      Produced record to topic test1 partition [0] @ offset 4
+      Produced record to topic test1 partition [0] @ offset 5
+      Produced record to topic test1 partition [0] @ offset 6
+      Produced record to topic test1 partition [0] @ offset 7
+      Produced record to topic test1 partition [0] @ offset 8
+      Produced record to topic test1 partition [0] @ offset 9
+      10 messages were produced to topic test1
+      ...
+
+#. View the :devx-examples:`producer
```
Comment (ybyzek) [inline clients/docs/java.rst]: @tikimims Unnecessary line break with below?

File: clients/docs/c.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+      Message delivered to test1 [0] at offset 2 in 22.77ms: { "count": 3 }
+      Message delivered to test1 [0] at offset 3 in 22.78ms: { "count": 4 }
+      Message delivered to test1 [0] at offset 4 in 22.78ms: { "count": 5 }
+      Message delivered to test1 [0] at offset 5 in 22.78ms: { "count": 6 }
+      Message delivered to test1 [0] at offset 6 in 22.78ms: { "count": 7 }
+      Message delivered to test1 [0] at offset 7 in 22.79ms: { "count": 8 }
+      Message delivered to test1 [0] at offset 8 in 22.80ms: { "count": 9 }
+      Message delivered to test1 [0] at offset 9 in 22.81ms: { "count": 10 }
+      10/10 messages delivered
+
```
Comment (ybyzek) [inline clients/docs/c.rst:125]: Missing:

```
View the :devx-examples:`producer code|clients/cloud/c/producer.c`.
```

Same with consumer

File: clients/docs/java.rst [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+      Consumed record with key alice and value {"count":3}, and updated total count to 6
+      Consumed record with key alice and value {"count":4}, and updated total count to 10
+      Consumed record with key alice and value {"count":5}, and updated total count to 15
+      Consumed record with key alice and value {"count":6}, and updated total count to 21
+      Consumed record with key alice and value {"count":7}, and updated total count to 28
+      Consumed record with key alice and value {"count":8}, and updated total count to 36
+      Consumed record with key alice and value {"count":9}, and updated total count to 45
+
+#. When you are done, press ``Ctrl-C``.
+
```
Comment (ybyzek) [inline clients/docs/java.rst:142]: Missing link to consumer source code

File: clients/docs/java.rst [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+      [Running count]: alice, 15
+      [Running count]: alice, 21
+      [Running count]: alice, 28
+      [Running count]: alice, 36
+      [Running count]: alice, 45
+      ...
+
+#. When you are done, press ``Ctrl-C``.
+
+#. View the :devx-examples:`consumer
```
Comment (ybyzek) [inline clients/docs/java.rst]: Unnecessary line break?

Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
#. View the :devx-examples:`Kafka Streams 
```

File: clients/docs/c.rst [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+   - the local file with configuration parameters to connect to your Kafka
+     cluster
+   - the topic name
+
+   .. code-block:: bash
+
+      ./producer test1 $HOME/.confluent/librdkafka.config
+
+#. Verify the producer received all the messages. You should see:
```
Comment (ybyzek) [inline clients/docs/c.rst]:
```suggestion
#. Verify the producer sent all the messages. You should see:
```

File: clients/docs/c.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+      Received message on test1 [0] at offset 6: { "count": 7 }
+      User alice sum 28
+      Received message on test1 [0] at offset 7: { "count": 8 }
+      User alice sum 36
+      Received message on test1 [0] at offset 8: { "count": 9 }
+      User alice sum 45
+      Received message on test1 [0] at offset 9: { "count": 10 }
+      User alice sum 55
+
+#. Press ``Ctrl-C`` to exit.
```
Comment (ybyzek) [inline clients/docs/c.rst:168]: Missing link to consumer source code

**Comment by ybyzek** [comment]
@tikimims note https://github.com/tikimims/examples/pull/3 should be merged before we merge this one

File: clients/docs/java.rst [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+      Consumed record with key alice and value {"count":6}, and updated total count to 21
+      Consumed record with key alice and value {"count":7}, and updated total count to 28
+      Consumed record with key alice and value {"count":8}, and updated total count to 36
+      Consumed record with key alice and value {"count":9}, and updated total count to 45
+
+#. When you are done, press ``Ctrl-C``.
+
+
+Kafka Streams
+-------------
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
-------------
~~~~~~~~~~~~~
```

File: clients/docs/java.rst [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+      # Run the Avro consumer
+      mvn exec:java -Dexec.mainClass="io.confluent.examples.clients.cloud.ConsumerAvroExample" \
+      -Dexec.args="$HOME/.confluent/java.config test2"
+
+#. View the :devx-examples:`consumer Avro
+   code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/ConsumerAvroExample.java`.
+
+
+Avro Kafka Streams
+------------------
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
~~~~~~~~~~~~~~~~~~
```

**Review by ybyzek — CHANGES_REQUESTED** [review]

File: clients/docs/java.rst (hunk 1 above)
Comment (tikimims) [reply clients/docs/java.rst]: Oh sorry —just a styling preference on my end. I can remove that. :-)

File: clients/docs/c.rst (hunk 2 above)
Comment (tikimims) [reply clients/docs/c.rst:125]: Correcto. How did I forget that smh lol

File: clients/docs/java.rst (hunk 3 above)
Comment (tikimims) [reply clients/docs/java.rst:142]: Yep

File: clients/docs/java.rst (hunk 4 above)
Comment (tikimims) [reply clients/docs/java.rst]: Oh sorry —just a styling preference on my end. I can remove that. :-)

File: clients/docs/java.rst (hunk 1 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 3 above)
Comment (tikimims) [reply clients/docs/java.rst:142]: Updated

File: clients/docs/java.rst (hunk 8 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 7 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 4 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/c.rst (hunk 6 above)
Comment (tikimims) [reply clients/docs/c.rst:168]: Added

File: clients/docs/c.rst (hunk 2 above)
Comment (tikimims) [reply clients/docs/c.rst:125]: Added

File: clients/docs/c.rst [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+Each record has a key representing a username (for example, ``alice``) and a
+value of a count, in JSON format (for example, ``{"count": 0}``). The consumer
+reads the same topic and keeps a rolling sum of the counts as it processes each
+record.
+
+
+Produce Records
+~~~~~~~~~~~~~~~
+
+#. Run the producer, passing in arguments for:
```
Comment (ybyzek) [inline clients/docs/c.rst:87]: Prior to running the producer, users need to build the project:

```
#. Build the example `producer` and `consumer` applications.

   .. code-block:: bash
      make

   Your output should resemble:

   .. code-block:: bash

      cc   consumer.c common.c json.c -o consumer -lrdkafka -lm
      cc   producer.c common.c json.c -o producer -lrdkafka -lm
```

File: clients/docs/c.rst [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+     cluster
+   - the topic name
+
+   .. code-block:: bash
+
+      ./producer test1 $HOME/.confluent/librdkafka.config
+
+#. Verify the producer sent all the messages. You should see:
+
+   .. code-block:: bash
```
Comment (ybyzek) [inline clients/docs/c.rst]: The format of this code-block is rendering strangely.  For consideration: should we change language from `bash`  to `none`?

Screen Shot 2020-08-07 at 08 33 48 AM

File: clients/docs/c.rst [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+reads the same topic and keeps a rolling sum of the counts as it processes each
+record.
+
+
+Produce Records
+~~~~~~~~~~~~~~~
+
+#. Run the producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka
```
Comment (ybyzek) [inline clients/docs/c.rst:89]: For this example, we need to switch the order of these two bullet points (first `the topic name`, second `the local file...`) to match the code

File: clients/docs/c.rst [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+      10/10 messages delivered
+
+#. View the :devx-examples:`producer code|clients/cloud/c/producer.c`.
+
+Consume Records
+~~~~~~~~~~~~~~~
+
+#. Run the consumer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka
```
Comment (ybyzek) [inline clients/docs/c.rst:134]: For this example, we need to switch the order of these two bullet points (first `the topic name`, second `the local file...`) to match the code

File: clients/docs/c.rst [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+     cluster.
+   - the topic name you used earlier
+
+   .. code-block:: bash
+
+      ./consumer test1 $HOME/.confluent/librdkafka.config
+
+#. Verify the consumer received all the messages. You should see:
+
+   .. code-block:: bash
```
Comment (ybyzek) [inline clients/docs/c.rst]: Same formatting issue here -- change language to `none`?

**Review by ybyzek — CHANGES_REQUESTED** [review]

File: clients/docs/java.rst [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+~~~~~~~~~~~~~~~
+
+#. Run the producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
```
Comment (ybyzek) [inline clients/docs/java.rst:71]: Let's make Compile the java code and `mvn clean package` its own numbered bullet.

So, above, before, `Run the producer...` add

```
#. Compile the java code

    .. code-block:: bash

      mvn clean package
```

Then you can remove all instances of this here, and below.

Comment (ybyzek) [inline clients/docs/java.rst]: Let's make Compile the java code and `mvn clean package` its own numbered bullet.

So, above, before, `Run the producer...` add

```
#. Compile the java code

   .. code-block:: bash

      mvn clean package
```

Then you can remove all instances of this here, and below.

File: clients/docs/java.rst [hunk 15]
```diff
@@ -0,0 +1,317 @@
...
+      mvn clean package
+
+      # Run the producer
+          # If the topic does not already exist, the code will use the Kafka Admin Client API to create the topic
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
      # If the topic does not already exist, the code will use the Kafka Admin Client API to create the topic
```

File: clients/docs/java.rst [hunk 16]
```diff
@@ -0,0 +1,317 @@
...
+~~~~~~~~~~~~~~~
+
+#. Run the consumer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name you used earlier
+
+   .. code-block:: text
+
+      # Compile the Java code
```
Comment (ybyzek) [inline clients/docs/java.rst]: Next 3 lines are now eligible for deletion (assuming you add the distinct compile step above)

File: clients/docs/java.rst [hunk 17]
```diff
@@ -0,0 +1,317 @@
...
+
+#. Run the Kafka Streams application, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka
+     cluster
+   - the topic name you used earlier
+
+   .. code-block:: text
+
+      # Compile the Java code
```
Comment (ybyzek) [inline clients/docs/java.rst]: Next 3 lines are now eligible for deletion (assuming you add the distinct compile step above)

File: clients/docs/java.rst [hunk 18]
```diff
@@ -0,0 +1,317 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro producer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
```
Comment (ybyzek) [inline clients/docs/java.rst]: Next 3 lines are now eligible for deletion (assuming you add the distinct compile step above)

File: clients/docs/java.rst [hunk 19]
```diff
@@ -0,0 +1,317 @@
...
+~~~~~~~~~~~~~~~~~~~~
+
+#. Run the Avro consumer, passing in arguments for:
+
+   - the local file with configuration parameters to connect to your Kafka cluster
+   - the topic name
+
+   .. code-block:: text
+
+      # Compile the Java code
```
Comment (ybyzek) [inline clients/docs/java.rst]: Next 3 lines are now eligible for deletion (assuming you add the distinct compile step above)

File: clients/docs/java.rst [hunk 20]
```diff
@@ -0,0 +1,317 @@
...
+
+#. Run the Avro Kafka Streams application, passing in arguments for:
+
+   -  the local file with configuration parameters to connect to your Kafka
+      cluster
+   -  the same topic name you used earlier
+
+   .. code-block:: text
+
+      # Compile the Java code
```
Comment (ybyzek) [inline clients/docs/java.rst]: Next 3 lines are now eligible for deletion (assuming you add the distinct compile step above)

File: clients/docs/java.rst [hunk 21]
```diff
@@ -0,0 +1,317 @@
...
+
+#. View the :devx-examples:`Kafka Streams code|clients/cloud/java/src/main/java/io/confluent/examples/clients/cloud/StreamsExample.java`.
+
+
+Avro and Confluent Cloud Schema Registry
+-----------------------------------------
+
+This example is similar to the previous example, except the value is formatted
+as Avro and integrates with the |ccloud| |sr|. Before using |ccloud| |sr|, check
+its |ccloud| |sr|.
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
its :ref:`availability and limits <cloud-limits>`.
```

File: clients/docs/java.rst [hunk 22]
```diff
@@ -0,0 +1,317 @@
...
+
+      # View the schema information for subject `test2-value`
+      curl -u {{ SR_API_KEY }}:{{ SR_API_SECRET }} https://{{ SR_ENDPOINT }}/subjects/test2-value/versions/1
+      {"subject":"test2-value","version":1,"id":100001,"schema":"{\"name\":\"io.confluent.examples.clients.cloud.DataRecordAvro\",\"type\":\"record\",\"fields\":[{\"name\":\"count\",\"type\":\"long\"}]}"}
+
+#. For schema evolution, you can `test schema
+   compatibility
+   <https://docs.confluent.io/current/schema-registry/develop/maven-plugin.html?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud#schema-registry-test-compatibility>`__
+   between newer schema versions and older schema versions in |ccloud| |sr|. The
+   `pom.xml <pom.xml>`__ hardcodes the Schema Registry subject name to
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
   :devx-examples:`pom.xml|clients/cloud/java/pom.xml` hardcodes the Schema Registry subject name to
```

File: clients/docs/java.rst [hunk 23]
```diff
@@ -0,0 +1,317 @@
...
+   ``test2-value``—change this if you didn't use topic name ``test2``. Then
+   test local schema compatibility for
+   `src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc
```
Comment (ybyzek) [inline clients/docs/java.rst]: Next 6 lines should be replaced to the following

```
:devx-example:`DataRecordAvro2a.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc`, which should fail, and
:devx-example:`DataRecordAvro2b.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2b.avsc`, which should pass.
```

File: clients/docs/java.rst [hunk 24]
```diff
@@ -0,0 +1,317 @@
...
+
+   .. code-block:: text
+
+      # DataRecordAvro2a.avsc compatibility test: FAIL
+      mvn schema-registry:test-compatibility "-DschemaRegistryUrl=https://{{ SR_ENDPOINT }}" "-DschemaRegistryBasicAuthUserInfo={{ SR_API_KEY }}:{{ SR_API_SECRET }}" "-DschemaLocal=src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc"
+
+      # DataRecordAvro2b.avsc compatibility test: PASS
+      mvn schema-registry:test-compatibility "-DschemaRegistryUrl=https://{{ SR_ENDPOINT }}" "-DschemaRegistryBasicAuthUserInfo={{ SR_API_KEY }}:{{ SR_API_SECRET }}" "-DschemaLocal=src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2b.avsc"
+
+#. View the :devx-examples:`Kafka Streams
```
Comment (ybyzek) [inline clients/docs/java.rst]: These two lines should be deleted

File: clients/docs/java.rst [hunk 25]
```diff
@@ -0,0 +1,317 @@
...
+
+This example is similar to the previous example, except the value is formatted
+as Avro and integrates with the |ccloud| |sr|. Before using |ccloud| |sr|, check
+its |ccloud| |sr|.
+
+#. .. include:: includes/client-example-vpc.rst
+
+#. .. include:: includes/client-example-schema-registry-1.rst
+
+#. .. include:: includes/client-example-schema-registry-2.rst
```
Comment (ybyzek) [inline clients/docs/java.rst]: Unfortunately we need a new include and the reference should be replaced in this example.  

1. Can you please create `includes/client-example-schema-registry-2-java.rst`, and in it, replace `librdkafka.config` with `java.config`?
2. Then in this file, replace `includes/client-example-schema-registry-2.rst` with `includes/client-example-schema-registry-2-java.rst`

File: clients/docs/java.rst [hunk 26]
```diff
@@ -0,0 +1,317 @@
...
+#. Change directory to the example for Java.
+
+   .. code-block:: bash
+
+      cd clients/cloud/java/
+
+#. .. include:: includes/client-example-create-file-java.rst
+
+
+Basic Producer and Consumer
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
Basic Producer and Consumer and Kafka Streams
```

**Review by ybyzek — CHANGES_REQUESTED** [review]

File: clients/docs/c.rst (hunk 9 above)
Comment (tikimims) [reply clients/docs/c.rst:87]: added

File: clients/docs/c.rst (hunk 10 above)
Comment (tikimims) [reply clients/docs/c.rst]: I've changed it to ".. code-block:: text" should be good now :-)

File: clients/docs/c.rst (hunk 11 above)
Comment (tikimims) [reply clients/docs/c.rst:89]: updated

File: clients/docs/c.rst (hunk 12 above)
Comment (tikimims) [reply clients/docs/c.rst:134]: updated

File: clients/docs/c.rst (hunk 13 above)
Comment (tikimims) [reply clients/docs/c.rst]: updated

File: clients/docs/java.rst (hunk 14 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 26 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 21 above)
Comment (tikimims) [reply clients/docs/java.rst]: updated

File: clients/docs/java.rst (hunk 25 above)
Comment (tikimims) [reply clients/docs/java.rst]: Added/updated

File: clients/docs/java.rst (hunk 22 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 23 above)
Comment (tikimims) [reply clients/docs/java.rst]: Updated

File: clients/docs/java.rst (hunk 24 above)
Comment (tikimims) [reply clients/docs/java.rst]: removed

File: clients/docs/java.rst [hunk 27]
```diff
@@ -0,0 +1,337 @@
...
+      {"subject":"test2-value","version":1,"id":100001,"schema":"{\"name\":\"io.confluent.examples.clients.cloud.DataRecordAvro\",\"type\":\"record\",\"fields\":[{\"name\":\"count\",\"type\":\"long\"}]}"}
+
+#. For schema evolution, you can `test schema
+   compatibility
+   <https://docs.confluent.io/current/schema-registry/develop/maven-plugin.html?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud#schema-registry-test-compatibility>`__
+   between newer schema versions and older schema versions in |ccloud| |sr|. The
+   :devx-examples:`pom.xml|clients/cloud/java/pom.xml` hardcodes the Schema
+   Registry subject name to ``test2-value``—change this if you didn't use topic
+   name ``test2``. Then test local schema compatibility for
+   :devx-example:`DataRecordAvro2a.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc`,
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
   :devx-examples:`DataRecordAvro2a.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2a.avsc`,
```

File: clients/docs/java.rst [hunk 28]
```diff
@@ -0,0 +1,337 @@
...
+   which should fail, and
+   :devx-example:`DataRecordAvro2b.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2b.avsc`,
```
Comment (ybyzek) [inline clients/docs/java.rst]:
```suggestion
   :devx-examples:`DataRecordAvro2b.avsc|clients/cloud/java/src/main/resources/avro/io/confluent/examples/clients/cloud/DataRecordAvro2b.avsc`,
```

File: clients/docs/c.rst [hunk 29]
```diff
@@ -0,0 +1,170 @@
+.. _client-examples-c:
+
+C
+===
```
Comment (ybyzek) [inline clients/docs/c.rst]:
```suggestion
=
```

File: clients/docs/c.rst [hunk 30]
```diff
@@ -0,0 +1,170 @@
...
+-------------
+
+Client
+~~~~~~
+
+-  `librdkafka <https://github.com/edenhill/librdkafka>`__ installed on
+   your machine. See the `librdkafka installation
+   instructions <https://github.com/edenhill/librdkafka/blob/master/README.md#instructions>`__.
+
+-  Create a local file (for example, at ``$HOME/.confluent/librdkafka.config``)
```
Comment (ybyzek) [inline clients/docs/c.rst:23]: @tikimims this bullet should be deleted (it's covered by the include in Kafka Cluster section)

**Review by ybyzek — APPROVED** [review]
@tikimims LGTM and local builds pass

**Pull request merged**